### PR TITLE
Bugfix#683 storagebuilders init

### LINKED ISSUE #683: Setting elevation : false while still building HillIndex storage breaks creation of other following extended storages.
The function
```public void processEdge(ReaderWay way, EdgeIteratorState edge, Coordinate[] coords) {
		if(arrStorageBuilders != null) {
			int nStorages = arrStorageBuilders.length;
			for(int i=0; i<nStorages; i++) {
				arrStorageBuilders[i].processEdge(way, edge, coords);
			}
		}
	}
```
in GraphProcessContext contains a HillIndexStorage at arrStorageBuilder[0] if elevation is set to false in the app.config, while the HillIndexStorage is indicated in the extended storages. 
The init of this storage is potentially not (correctly) called and results in a HillIndexGraphStorageBuilder that has a field hillIndexCalc == null.
This leads to an early termination of the above function, which is caught with only a warning in the execution of the above code block. As the HillIndexStorageBuilder is at array position 0, none of the following StorageBuilders are called, while no proper exception is being thrown.
Removing the HillIndexStorage from the extended storages in the app.config solves the problem, but the failing combination should be caught in the code.
The init of the storage should in theory already return a null Builder for this storage given no elevation, but it does not seem to work. Maybe something broken with graphhopper.hasElevation()?
Potentially due to app.config parameter  ` elevation_provider: multi `being set

### PR DESCRIPTION
**Pull request by takb**
### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the master branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [x] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

Fixes #683  .

### Information about the changes
- Reason for change:
Correctly handle the HillIndexStorage not being able to initialize (if elevation data missing). Also added checks and feedback during ExtraInfoProcessor instantiation. 
Also fixed: Setting the elevation parameter to false for a profile was ignored if previous profiles had it enabled,

### CODE REVIEW

File: openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfile.java [hunk 1]
```diff
@@ -176,7 +176,9 @@ public static ORSGraphHopper initGraphHopper(String osmFile, RouteProfileConfigu
         // so the caching for elevation data will/can be reused across different vehicles. [the loadCntx is a single
         // Object that will shared across the (potential) multiple running instances]
         if(loadCntx.getElevationProvider() != null) {
-            gh.setElevationProvider(loadCntx.getElevationProvider());
+            if (args.has("graph.elevation.provider")) {
```
Comment (HendrikLeuschner) [inline openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfile.java:179]: What happens if `(loadCntx.getElevationProvider() != null) && !args.has("graph.elevation.provider")`

Comment (takb) [reply openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfile.java:179]: the loadcontext is used to store elevationprovider if it has been instantiated before, as far as i understand to make enable caching of data, and the problem was that every later profile was assigned the provider regardless if it had elevation parameter set to true. in the case in question the next line is supposed to not run.

File: openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java [hunk 2]
```diff
@@ -141,6 +140,7 @@ protected void cleanUp() {
 			if (LOGGER.isInfoEnabled())
 				LOGGER.info(String.format("will create PrepareRoutingSubnetworks with:%n\tNodeCountBefore: '%d'%n\tgetAllEdges().getMaxId(): '%d'%n\tList<FlagEncoder>: '%s'%n\tminNetworkSize: '%d'%n\tminOneWayNetworkSize: '%d'", prevNodeCount, ex, list, minNetworkSize, minOneWayNetworkSize)
 			);
+			ghs.getProperties().put("elevation", hasElevation());
```
Comment (HendrikLeuschner) [inline openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java:143]: What is this used for? In the HillIndexStorage, graphhopper.hasElevation() is called directly.

**Review by HendrikLeuschner — APPROVED** [review]
Looks fine to me

**Pull request merged**